core/src/main/kotlin/org/jetbrains/kotlinx/dataframe/api/first.kt

```diff
@@ -27,32 +30,185 @@ import kotlin.reflect.KProperty
 
 // region DataColumn
 
+/**
+ * Returns the first value in this [DataColumn].
+ *
+ * @see [firstOrNull]
+ * @see [last]
+ * @see [take]
+ * @see [takeLast]
+ *
+ * @return The first value in this [DataColumn].
+ *
+ * @throws [IndexOutOfBoundsException] if the [DataColumn] is empty.
+ */
 public fun <T> DataColumn<T>.first(): T = get(0)
 
+/**
+ * Returns the first value in this [DataColumn]. If the [DataColumn] is empty, returns `null`.
+ *
+ * @see [first]
+ * @see [last]
+ * @see [take]
+ * @see [takeLast]
+ *
+ * @return The first value in this [DataColumn], or `null` if the [DataColumn] is empty.
+ */
 public fun <T> DataColumn<T>.firstOrNull(): T? = if (size > 0) first() else null
 
+/**
+ * Returns the first value in this [DataColumn] that matches the given [predicate].
+ *
+ * ### Example
+ * ```kotlin
+ * // In a DataFrame of financial transactions sorted by time,
+ * // find the amount of the first transaction over 100 euros
+ * df.amount.first { it > 100 }
+ * ```
+ *
+ * @see [firstOrNull]
+ * @see [last]
+ * @see [take]
+ * @see [takeLast]
+ *
+ * @param [predicate] A lambda expression used to get the first value
+ * that satisfies a condition specified in this expression.
+ * This predicate takes a value from the [DataColumn] as an input
+ * and returns `true` if the value satisfies the condition or `false` otherwise.
+ *
+ * @return The first value in this [DataColumn] that matches the given [predicate].
+ *
+ * @throws [NoSuchElementException] if the [DataColumn] contains no elements matching the [predicate]
+ * (including the case when the [DataColumn] is empty).
+ */
 public fun <T> DataColumn<T>.first(predicate: (T) -> Boolean): T = values.first(predicate)
 
+/**
+ * Returns the first value in this [DataColumn] that matches the given [predicate].
+ * Returns `null` if the [DataColumn] contains no elements matching the [predicate]
+ * (including the case when the [DataColumn] is empty).
+ *
+ * ### Example
+ * ```kotlin
+ * // In a DataFrame of financial transactions sorted by time,
+ * // find the amount of the first transaction over 100 euros,
+ * // or 'null' if there is no such transaction
+ * df.amount.firstOrNull { it > 100 }
+ * ```
+ *
+ * @see [first]
+ * @see [last]
+ * @see [take]
+ * @see [takeLast]
+ *
+ * @param [predicate] A lambda expression used to get the first value
+ * that satisfies a condition specified in this expression.
+ * This predicate takes a value from the [DataColumn] as an input
+ * and returns `true` if the value satisfies the condition or `false` otherwise.
+ *
+ * @return The first value in this [DataColumn] that matches the given [predicate],
+ * or `null` if the [DataColumn] contains no elements matching the [predicate].
+ */
 public fun <T> DataColumn<T>.firstOrNull(predicate: (T) -> Boolean): T? = values.firstOrNull(predicate)
 
 // endregion
 
 // region DataFrame
 
+/**
+ * Returns the first [row][DataRow] in this [DataFrame].
+ *
+ * @see [DataFrame.firstOrNull]
```

Comment:
Do you actually need to provide receiver here?
Anyway, please, omit it. Just `[firstOrNull]` should work, but if not, please use 
`[firstOrNull][DataFrame.firstOrNull]`